From ab5388ac050be63a564b58809ecca3cb1269c3d0 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Wed, 19 May 2021 19:05:49 +0800
Subject: [PATCH] msdk: fix includedir path backport from poky

Here is commit id from Poky:
commit c750a47a5e98e768e68b1f15f524b12173fba685
Author: Naveen Saini <naveen.kumar.saini@intel.com>
Date:   Wed Jan 13 11:07:22 2021 +0800

    gstreamer1.0-plugins-bad: fix msdk pkgconfig build failure

    Error log:
    | Run-time dependency libmfx found: YES 1.34
    | Has header "mfx/mfxdefs.h" : YES
    |
    | ../gst-plugins-bad-1.18.2/sys/msdk/meson.build:75:2: ERROR: Include dir /usr/include/mfx does not exist.
    |
    | A full log can be found at /home/s

    When msdk is enable in PACKAGECONFIG, plugin fails to locate includedir, so
    adding a patch which prepend PKG_CONFIG_SYSROOT_DIR to the dir path

    (From OE-Core rev: 204aa9cfa6ac99846ac494778dd143a13746ac7e)

    Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
    Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 sys/msdk/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sys/msdk/meson.build b/sys/msdk/meson.build
index 9e95973f2..a74720d8c 100644
--- a/sys/msdk/meson.build
+++ b/sys/msdk/meson.build
@@ -46,7 +46,9 @@ if mfx_api != 'oneVPL'
   mfx_dep = dependency('libmfx', version: ['>= 1.0', '<= 1.99'], required: false)
 
   if mfx_dep.found()
+    pkgconf_sysroot = run_command(python3, '-c', 'import os; print(os.environ.get("PKG_CONFIG_SYSROOT_DIR"))').stdout().strip()
     mfx_incdir = mfx_dep.get_pkgconfig_variable('includedir')
+    mfx_incdir = pkgconf_sysroot +  mfx_incdir
     mfx_inc = []
     use_msdk = true
   else
-- 
2.32.0

